store/store.go

```diff
@@ -1157,6 +1157,9 @@ func (s *Store) Query(qr *proto.QueryRequest) ([]*proto.QueryRows, error) {
 			return nil, err
 		}
 
+		if qr.LeaderReadOpt {
```

Comment:
I would structure this code differently. In the event that the "leader read" is selected, you/ve done a compression in the lines about for no good reason.

I'll add more comments in the context.

Reply:
Change:
`if qr.Level == proto.QueryRequest_QUERY_REQUEST_LEVEL_STRONG`
to
`if qr.Level == proto.QueryRequest_QUERY_REQUEST_LEVEL_STRONG && !qr.LeaderReadOpt()`

then, add the very end of the function add this:

```Go
if qr.LeaderReadOpt() {
    if err := s.verifyLeader(); err != nil {
        return nil, err
    }
}
return s.db.Query(qr.Request, qr.Timings)
```
`s.verifyLeader()` will need to be written, it will just call the Raft code, not do any query.